### PR TITLE
Feature multi measure

### PR DESCRIPTION
**Pull request by kafnevod**
Add support drawing several measured figures on the map and
button + for enabling last measure instrument.

### CODE REVIEW

File: addon/components/flexberry-maptoolbar.js [hunk 1]
```diff
@@ -13,11 +13,20 @@ export default Ember.Component.extend({
 
   activeToolName: null,
 
+  _isMeasureTool(toolname) {
+    return toolname && toolname.substr(-11) ===  'measuretool';
```
Comment (kuzkok) [inline addon/components/flexberry-maptoolbar.js]: it is a hack, bad solution. Replace this with property in MapTool class thats indicate what tool can be enabled many times

File: addon/components/flexberry-maptoolbar.js [hunk 2]
```diff
@@ -13,11 +13,20 @@ export default Ember.Component.extend({
...
+  },
+
   actions: {
 
     activateTool(toolname) {
-      if(toolname === this.get('activeToolName')) {
-        return;
+      let activeToolName = this.get('activeToolName');
+      if (toolname === '+' && this._isMeasureTool(activeToolName)) {
```
Comment (kuzkok) [inline addon/components/flexberry-maptoolbar.js]: toolname === '+' ?

File: addon/maptools/rectanglemeasuretool.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import MapTool from 'ember-flexberry-gis/maptools/maptool';
+
+export default MapTool.extend({
+
+  _layer: null,
+  _measure: null,
+
+  rectangleClose(e) {
```
Comment (kuzkok) [inline addon/maptools/rectanglemeasuretool.js]: where this function is used? if it not used - should be deleted

File: tests/dummy/app/templates/map.hbs [hunk 4]
```diff
@@ -20,6 +20,22 @@
           <i class="info circle icon"></i>
         </button>
       </div>
+      <div class='ui compact menu' style='border-width:0px;border-image-width:0px;box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px 1px inset; hight:36px;'>
```
Comment (kuzkok) [inline tests/dummy/app/templates/map.hbs]: inline styles - bad practice, should be in css

File: tests/dummy/app/templates/map.hbs [hunk 5]
```diff
@@ -20,6 +20,22 @@
...
+        <div class='ui simple dropdown item'  style='background-color:rgb(224, 225, 226);color: rgba(0, 0, 0, 0.6);font-weight: bold;padding:8px;height:36px'>
+        <i class='wizard icon'></i>Измерить
+        <button class="ui icon button" {{action "activateTool" "+" target=toolbar}}  title='Добавить текущий инструмент измерения' style=' hight:36px;'>
+        <i class="plus icon"></i>
+        </button>
+        <div class='menu'   style='z-index:2000 !important'>
+            <a class='item' {{action "activateTool" "markermeasuretool" target=toolbar}}><i class='marker icon'></i>Маркер</a>
```
Comment (kuzkok) [inline tests/dummy/app/templates/map.hbs]: all labels should be i18zed

File: addon/components/flexberry-maptoolbar.js (hunk 2 above)
Comment (kafnevod) [inline addon/components/flexberry-maptoolbar.js]: If i want to draw same (current) measure tool after cliking +-button it send toolname +

Comment (kafnevod) [inline addon/components/flexberry-maptoolbar.js]: Remove button and support

File: addon/components/flexberry-maptoolbar.js (hunk 1 above)
Comment (kafnevod) [inline addon/components/flexberry-maptoolbar.js]: For all measuredtools add superclass measuredtool with common methods: isMultiTool()=true, enable(), disable(), measured(), measuredClose()
Add method isMultiTool()=false to maptool

File: tests/dummy/app/templates/map.hbs (hunk 4 above)
Comment (kafnevod) [inline tests/dummy/app/templates/map.hbs]: I think we should create flexberry-mapmenu for this.
I have'nt time and skills to do it

File: tests/dummy/app/templates/map.hbs (hunk 5 above)
Comment (kafnevod) [inline tests/dummy/app/templates/map.hbs]: Done
We should discuss how to localize Leaflet.Editable.Measures classes

File: addon/maptools/maptool.js [hunk 6]
```diff
@@ -5,6 +5,10 @@ export default Ember.Object.extend({
 
   map: null,
 
+  isMultiTool() {
+    return false;
+  },
```
Comment (kuzkok) [inline addon/maptools/maptool.js]: for which purposes it is method? Why not property?

File: addon/maptools/measuretool.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+export default MapTool.extend({
+  _layer: null,
+  _measure: null,
+
+  isMultiTool() {
+    return true;
+  },
+
+  measureClose(e) {
```
Comment (kuzkok) [inline addon/maptools/measuretool.js]: if variable isnt used - should be deleted

File: addon/maptools/measuretool.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    this._layer = e.layer;
+    this.get('map').off('measure:created');
+  },
+
+  enable() {
+    this._super(...arguments);
+    // there are problems if dragging is disabled
+    var map = this.get('map');
+    map.dragging.enable();
+    this._measure = map.MeasureTools ? map.MeasureTools : L.measureBase(map);
```
Comment (kuzkok) [inline addon/maptools/measuretool.js]: this map.MeasureTools is never be initialized, why you check it here?

File: addon/components/flexberry-maptoolbar.js [hunk 9]
```diff
@@ -11,33 +11,34 @@ export default Ember.Component.extend({
 
   activeTool: null,
 
-  activeToolName: null,
+//   activeToolName: null,
+
+  _isMeasureTool(toolname) {
+    return toolname && toolname.substr(-11) ===  'measuretool';
```
Comment (kuzkok) [inline addon/components/flexberry-maptoolbar.js]: You should remove this method., And unneccesary comment two lines above.

File: addon/maptools/measuretool.js (hunk 8 above)
Comment (kafnevod) [inline addon/maptools/measuretool.js]: map.MeasureTools is initialized in L.measureBase(map) (similar as properties map.editTools in Leaflet.Measure)

File: addon/maptools/maptool.js (hunk 6 above)
Comment (kafnevod) [inline addon/maptools/maptool.js]: In my opinion on class inheritance it's better methods overriding, as properties overriding and reinitializing.

File: addon/maptools/measuretool.js (hunk 7 above)
Comment (kafnevod) [inline addon/maptools/measuretool.js]: Do you mean _layer, _measure
Deleted on disable()?
Deleted - set to null or undefined?

Comment (kuzkok) [inline addon/maptools/measuretool.js]: *unused, its about function parameter `e`

File: addon/maptools/maptool.js (hunk 6 above)
Comment (SeregaMatin) [inline addon/maptools/maptool.js]: I agree, here it shold be a property.

File: addon/locales/en/translations.js [hunk 10]
```diff
@@ -137,6 +137,14 @@ export default {
     },
 
     'flexberry-maplayer': {
+    },
+    'flexberry-maptoolbar': {
+      'measureCaption': 'Measure',
+      'mearkerMeasure': 'Marker',
```
Comment (SeregaMatin) [inline addon/locales/en/translations.js]: Not 'Marker', but 'Coordinates'.

File: addon/locales/en/translations.js [hunk 11]
```diff
@@ -137,6 +137,14 @@ export default {
...
+      'circleMeasure': 'Circle',
```
Comment (SeregaMatin) [inline addon/locales/en/translations.js]: Not 'Circle', but 'Radius'.

File: addon/locales/en/translations.js [hunk 12]
```diff
@@ -137,6 +137,14 @@ export default {
...
+      'rectangleMeasure': 'Rectangle',
+      'polylineMeasure': 'Polyline',
```
Comment (SeregaMatin) [inline addon/locales/en/translations.js]: Not 'Polyline', but 'Distance'.

File: addon/locales/en/translations.js [hunk 13]
```diff
@@ -137,6 +137,14 @@ export default {
...
+      'polygonMeasure': 'Polygon'
```
Comment (SeregaMatin) [inline addon/locales/en/translations.js]: Not 'Polygon', but 'Area'.

File: tests/dummy/app/templates/map.hbs (hunk 4 above)
Comment (SeregaMatin) [inline tests/dummy/app/templates/map.hbs]: I think we should wrap every maptool into component, but it is separate task
But for a while, move styles into CSS, please, into ./addon/styles/maptools/measure.scss and import this style inside ./addon/styles/addon.scss.

**Pull request closed without merging**